cloudcore panic on delete configmap

<!-- Please use this template while reporting a bug and provide as much info as possible. Thanks!-->
**What happened**:
when i delete configmap, cloudcore panic
```
19:49:21.081074 715791 upstream.go:92] Parse message: 55097b9c-35be-4e50-a2c5-3025b972b2a8 resource type with error: unknown resource
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x14bc8d3]
goroutine 25 [running]:
github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/controller.kubeClientGet(0xc0004ee1c0, 0xc0010fb04e, 0x7, 0xc0010fb060, 0x17, 0x182a761, 0x9, 0x0, 0xc0013e53e0, 0x5, ...)
        /data/gopath/src/github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/controller/upstream.go:507 +0x543
github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/controller.queryInner(0xc0004ee1c0, 0xc001b70510, 0x24, 0xc0014af2f0, 0x24, 0x1775d6d189a, 0x0, 0x0, 0x0, 0xc0013e53e0, ...)
        /data/gopath/src/github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/controller/upstream.go:559 +0x4d3
github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/controller.(*UpstreamController).queryConfigMap(0xc0004ee1c0)
        /data/gopath/src/github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/controller/upstream.go:598 +0xa3
created by github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/controller.(*UpstreamController).Start
        /data/gopath/src/github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/controller/upstream.go:134 +0x478
```


**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
when i delete configmap(not pod), cloudcore panic
**Anything else we need to know?**:

**Environment**:
<details><summary>Kubernets Version</summary>

```console
$ kubelet --version
[root@10-1-0-13 ~]# kubelet --version
Kubernetes v1.18.4-tke.6
$ kubectl version
[root@VM-0-12-centos ~]# kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.4-tke.6", GitCommit:"194201819cf1e5cf45d38f72ce1aac9efca4c7ff", GitTreeState:"clean", BuildDate:"2020-12-29T09:13:24Z", GoVersion:"go1.15.6", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

<details><summary>CloudCore Version</summary>

```console
$ cloudcore --version
[root@VM-0-12-centos jsonnet]# cloudcore --version
KubeEdge v1.5.0-beta.0.258+42678d7ca463fd
```

</details>

<details><summary>EdgeCore Version</summary>

```console
$ edgecore --version
[root@xxxx ~]# /etc/kubeedge/edgecore --version
2021-02-01 20:18:11.555298 I | INFO: Install client plugin, protocol: rest
2021-02-01 20:18:11.555481 I | INFO: Installed service discovery plugin: edge
KubeEdge v1.5.0
```

</details>

**CloudSide Environment**:
<details><summary>Hardware configuration</summary>

```console
$ lscpu
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              2
On-line CPU(s) list: 0,1
Thread(s) per core:  1
Core(s) per socket:  2
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               85
Model name:          Intel(R) Xeon(R) Platinum 8255C CPU @ 2.50GHz
Stepping:            5
CPU MHz:             2499.998
BogoMIPS:            4999.99
Hypervisor vendor:   KVM
Virtualization type: full
L1d cache:           32K
L1i cache:           32K
L2 cache:            4096K
L3 cache:            36608K
NUMA node0 CPU(s):   0,1
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 arat avx512_vnni
```

</details>

<details><summary>OS</summary>

```console
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="8 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="8"
```

</details>

<details><summary>Kernel</summary>

```console
$ uname -a
[root@VM-0-12-centos jsonnet]# uname -a
Linux VM-0-12-centos 4.18.0-80.el8.x86_64 #1 SMP Tue Jun 4 09:19:46 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

</details>

<details><summary>Go version</summary>

```console
$ go version
# paste output here
```

</details>

<details><summary>Others</summary>
</details>

**EdgeSide Environment**:
<details><summary>Hardware configuration</summary>

```console
$ lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                32
On-line CPU(s) list:   0-31
Thread(s) per core:    2
Core(s) per socket:    8
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 79
Model name:            Intel(R) Xeon(R) CPU E5-2620 v4 @ 2.10GHz
Stepping:              1
CPU MHz:               2299.876
BogoMIPS:              4199.77
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              20480K
NUMA node0 CPU(s):     0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30
NUMA node1 CPU(s):     1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a rdseed adx smap intel_pt xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts flush_l1d
```

</details>

<details><summary>OS</summary>

```console
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

```

</details>

<details><summary>Kernel</summary>

```console
$ uname -a
Linux hello-hostname 5.3.6-1.el7.elrepo.x86_64 #1 SMP Fri Oct 11 17:24:39 EDT 2019 x86_64 x86_64 x86_64 GNU/Linux
```

</details>

<details><summary>Go version</summary>

```console
$ go version
# paste output here
```

</details>

<details><summary>Others</summary>
</details>

--- a/cloud/pkg/edgecontroller/controller/upstream.go
+++ b/cloud/pkg/edgecontroller/controller/upstream.go
@@ -504,35 +504,59 @@ func kubeClientGet(uc *UpstreamController, namespace string, name string, queryT
 	switch queryType {
 	case model.ResourceTypeConfigmap:
 		configMap, err := uc.configMapLister.ConfigMaps(namespace).Get(name)
-		resourceVersion := configMap.ResourceVersion
+		resourceVersion := ""
+		if err == nil {
+			resourceVersion = configMap.ResourceVersion
+		}
 		return configMap, resourceVersion, err
 	case model.ResourceTypeSecret:
 		secret, err := uc.secretLister.Secrets(namespace).Get(name)
-		resourceVersion := secret.ResourceVersion
+		resourceVersion := ""
+		if err == nil {
+			resourceVersion = secret.ResourceVersion
+		}
 		return secret, resourceVersion, err
 	case common.ResourceTypeService:
 		svc, err := uc.serviceLister.Services(namespace).Get(name)
-		resourceVersion := svc.ResourceVersion
+		resourceVersion := ""
+		if err == nil {
+			resourceVersion = svc.ResourceVersion
+		}
 		return svc, resourceVersion, err
 	case common.ResourceTypeEndpoints:
 		eps, err := uc.endpointLister.Endpoints(namespace).Get(name)
-		resourceVersion := eps.ResourceVersion
+		resourceVersion := ""
+		if err == nil {
+			resourceVersion = eps.ResourceVersion
+		}
 		return eps, resourceVersion, err
 	case common.ResourceTypePersistentVolume:
 		pv, err := uc.kubeClient.CoreV1().PersistentVolumes().Get(context.Background(), name, metaV1.GetOptions{})
-		resourceVersion := pv.ResourceVersion
+		resourceVersion := ""
+		if err == nil {
+			resourceVersion = pv.ResourceVersion
+		}
 		return pv, resourceVersion, err
 	case common.ResourceTypePersistentVolumeClaim:
 		pvc, err := uc.kubeClient.CoreV1().PersistentVolumeClaims(namespace).Get(context.Background(), name, metaV1.GetOptions{})
-		resourceVersion := pvc.ResourceVersion
+		resourceVersion := ""
+		if err == nil {
+			resourceVersion = pvc.ResourceVersion
+		}
 		return pvc, resourceVersion, err
 	case common.ResourceTypeVolumeAttachment:
 		va, err := uc.kubeClient.StorageV1().VolumeAttachments().Get(context.Background(), name, metaV1.GetOptions{})
-		resourceVersion := va.ResourceVersion
+		resourceVersion := ""
+		if err == nil {
+			resourceVersion = va.ResourceVersion
+		}
 		return va, resourceVersion, err
 	case model.ResourceTypeNode:
 		node, err := uc.nodeLister.Get(name)
-		resourceVersion := node.ResourceVersion
+		resourceVersion := ""
+		if err == nil {
+			resourceVersion = node.ResourceVersion
+		}
 		return node, resourceVersion, err
 	default:
 		err := stderrors.New("Wrong query type")

